kernel/src/hil/screen.rs

```diff
@@ -0,0 +1,156 @@
+    /// that will return the actual ReturnCode after setting the rotation.
+    ///
+    /// note this can swap the width with height.
+    fn set_rotation(&self, rotation: ScreenRotation) -> ReturnCode;
+
+    /// Returns the number of the resolutions supported.
+    /// should return at least one (the current resolution)
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen (most screens do not support such a request,
+    /// resolutions are described in the data sheet).
+    ///
+    /// If the screen supports such a feature, the driver should request this information
+    /// from the screen upfront.
+    fn get_num_supported_resolutions(&self) -> usize;
+
+    /// Can be called with an index from 0 .. count-1 and will
+    /// a tuple (width, height) with the current resolution (in pixels).
+    /// note that width and height may change due to rotation
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen.
+    fn get_supported_resolution(&self, index: usize) -> Option<(usize, usize)>;
+
+    /// Returns the number of the pixel formats supported.
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen (most screens do not support such a request,
+    /// pixel formats are described in the data sheet).
+    ///
+    /// If the screen supports such a feature, the driver should request this information
+    /// from the screen upfront.
+    fn get_num_supported_pixel_formats(&self) -> usize;
+
+    /// Can be called with index 0 .. count-1 and will return
+    /// the value of each pixel format mode.
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen.
+    fn get_supported_pixel_format(&self, index: usize) -> Option<ScreenPixelFormat>;
+}
+
+pub trait Screen {
+    /// Returns a tuple (width, height) with the current resolution (in pixels)
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen.
+    ///
+    /// note that width and height may change due to rotation
+    fn get_resolution(&self) -> (usize, usize);
+
+    /// Returns the current pixel format
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen.
+    fn get_pixel_format(&self) -> ScreenPixelFormat;
+
+    /// Returns the current rotation.
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen.
+    fn get_rotation(&self) -> ScreenRotation;
+
+    /// Sets the video memory frame.
+    /// This function has to be called before the first call to the write function.
+    fn set_write_frame(&self, x: usize, y: usize, width: usize, height: usize) -> ReturnCode;
```

Comment:
```suggestion
    /// Sets the video memory frame.
    /// This function has to be called before the first call to the write function.
    /// This will generate a `command_complete()` callback when finished.
    ///
    /// Return values:
    /// - `SUCCESS`: The write frame is valid.
    /// - `EINVAL`: The parameters of the write frame are not valid.
    /// - `EBUSY`: Unable to set the write frame on the device.
    fn set_write_frame(&self, x: usize, y: usize, width: usize, height: usize) -> ReturnCode;
```